lua/telescope/builtin/files.lua

```diff
@@ -271,16 +271,14 @@ files.treesitter = function(opts)
   end
 
   local parsers = require "nvim-treesitter.parsers"
-  if not parsers.has_parser() then
+  if not parsers.has_parser(parsers.get_buf_lang(opts.bufnr)) then
     print "No parser for the current buffer"
     return
   end
 
   local ts_locals = require "nvim-treesitter.locals"
-  local bufnr = opts.bufnr or vim.api.nvim_get_current_buf()
-
   local results = {}
-  for _, definitions in ipairs(ts_locals.get_definitions(bufnr)) do
+  for _, definitions in ipairs(ts_locals.get_definitions(opts.bufnr)) do
```

Comment:
so prior to this `bufnr` was never nil now it can be if you just do `Telescope treesitter` or with `:lua ....`

Not sure if we want this. I am pretty sure we should do `opts.bufnr = vim.F.if_nil(opts.bufnr, vim.api.nvim_get_current_buf())`